### PR TITLE
Add INTEGER type to Presto

### PR DESCRIPTION
**Pull request by raghavsethi**
The last two commits will be squashed together before merge.

### CODE REVIEW

**Comment by arhimondr** [comment]
Are you still parsing integer literals as a `BIGINT` for a reason? Or you just missed that?

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java [hunk 1]
```diff
@@ -321,7 +326,7 @@ public long getLong(int fieldId)
     {
         checkState(!closed, "Cursor is closed");
 
-        if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
+        if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(INTEGER) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
             // we don't use Preconditions.checkArgument because it requires boxing fieldId, which affects inner loop performance
             throw new IllegalArgumentException(format("Expected field to be %s, %s or %s , actual %s (field %s)", BIGINT, DATE, TIMESTAMP, types[fieldId], fieldId));
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java]: `INTEGER` is missing as the allowed type in the message.

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java [hunk 2]
```diff
@@ -321,7 +326,7 @@ public long getLong(int fieldId)
     {
         checkState(!closed, "Cursor is closed");
 
-        if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
+        if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(INTEGER) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java]: Cannot we have it done similar to the approach from `ParquetPageSourceFactory` (with `Set.contains()`)? Or if performance concerns arise maybe we can at least extract this to a separate method (since you have the same check here and in `ColumnarTextHiveRecordCursor` and you definitely want it to be the same)?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 3]
```diff
@@ -254,6 +260,30 @@ public void createPartitionedTableAs(HiveStorageFormat storageFormat)
...
+                "FROM tpch.tiny.orders";
+
+        assertUpdate(createTable, "SELECT count(*) from orders");
+
+        List<HivePartition> partitions = getPartitions("test_create_partitioned_table_as");
+        assertEquals(partitions.size(), 3);
+
+        assertQuery("SELECT * from test_create_partitioned_table_as", "SELECT orderkey, CAST(shippriority AS INTEGER), orderstatus FROM orders");
+
+        assertUpdate("DROP TABLE test_create_partitioned_table_as");
```
Comment (maciejgrzybek) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: Can you put this in `finally` block (create table and `assertQuery` in `try`)? Just to ensure the cleanness of env after test run despite of it's outcome.

File: presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java [hunk 4]
```diff
@@ -52,10 +52,11 @@ public MaterializedRow(int precision, List<Object> values)
 
     private static Object processValue(int precision, Object value)
     {
+        checkArgument(!(value instanceof Integer || value instanceof Short || value instanceof Byte), "INTEGER, SMALLINT and TINYINT are not yet supported");
         if (value instanceof Double || value instanceof Float) {
             return new ApproximateDouble(((Number) value).doubleValue(), precision);
         }
-        if (value instanceof Number) {
+        if (value instanceof Long) {
             return ((Number) value).longValue();
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java]: You can replace this with `(long) value`

**Comment by cberner** [comment]
can you restart Travis? it looks like it broke after we moved this to prestodb project

File: presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java [hunk 5]
```diff
@@ -156,6 +157,13 @@ public static boolean castToBoolean(@SqlType(StandardTypes.DOUBLE) double value)
     }
 
     @ScalarOperator(CAST)
+    @SqlType(StandardTypes.INTEGER)
+    public static long castToInteger(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        return Ints.checkedCast((long) MathFunctions.round(value));
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java]: Add try-catch like above

File: presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        catch (ArithmeticException e) {
+            throw new PrestoException(DIVISION_BY_ZERO, e);
+        }
+    }
+
+    @ScalarOperator(NEGATION)
+    @SqlType(StandardTypes.INTEGER)
+    public static long negate(@SqlType(StandardTypes.INTEGER) long value)
+    {
+        return Math.negateExact((int) value);
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java]: try-catch

File: presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java [hunk 7]
```diff
@@ -215,6 +217,8 @@ private static boolean canCastTypeBase(String fromTypeBase, String toTypeBase)
             return true;
         }
         switch (fromTypeBase) {
+            case StandardTypes.INTEGER:
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java]: This sounds correct, but check with @martint that he doesn't see a problem with it

**Comment by cberner** [comment]
Is `INTEGER'1'` for int 1 and `1` for bigint standard SQL?

**Comment by raghavsethi** [comment]
Yep

File: presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java (hunk 6 above)
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java]: Shoot - sorry it looks like I messed up the commit order.. This is fixed in a later commit.

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java [hunk 8]
```diff
@@ -636,6 +641,9 @@ private void parseColumn(int column)
         else if (BIGINT.equals(type)) {
             parseLongColumn(column);
         }
+        else if (INTEGER.equals(type)) {
+            parseLongColumn(column);
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java]: it looks like there's no checking in `parseLongColumn` that a `HIVE_INT` is 4 bytes, so add a check there since that would be corrupt data

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java [hunk 9]
```diff
@@ -575,6 +580,9 @@ private void parseColumn(int column)
         else if (type.equals(BIGINT)) {
             parseLongColumn(column);
         }
+        else if (type.equals(INTEGER)) {
+            parseLongColumn(column);
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java]: same comment

File: presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java [hunk 10]
```diff
@@ -470,6 +475,9 @@ private void parseColumn(int column)
         else if (BIGINT.equals(type)) {
             parseLongColumn(column);
         }
+        else if (INTEGER.equals(type)) {
+            parseLongColumn(column);
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursor.java]: same

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java (hunk 8 above)
Comment (raghavsethi) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java]: According to the spec it should be: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types#LanguageManualTypes-IntegralTypes(TINYINT,SMALLINT,INT,BIGINT)

I'm concerned about the performance penalty of a check (or would branch prediction make this go away?).

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java [hunk 11]
```diff
@@ -295,11 +299,11 @@ private static Type getPrimitiveType(PrimitiveObjectInspector.PrimitiveCategory
             case BOOLEAN:
                 return BOOLEAN;
             case BYTE:
-                return BIGINT;
+                return INTEGER;
             case SHORT:
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java]: Are we planning to add SHORT and BYTE too?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 12]
```diff
@@ -365,7 +367,17 @@ public static NullableValue parsePartitionValue(String partitionName, String val
...
 
-            if (HIVE_BYTE.equals(hiveType) || HIVE_SHORT.equals(hiveType) || HIVE_INT.equals(hiveType) || HIVE_LONG.equals(hiveType)) {
+            if (HIVE_BYTE.equals(hiveType) || HIVE_SHORT.equals(hiveType) || HIVE_INT.equals(hiveType)) {
+                if (isNull) {
+                    return NullableValue.asNull(INTEGER);
+                }
+                if (value.isEmpty()) {
+                    return NullableValue.of(INTEGER, 0L);
+                }
+                return NullableValue.of(INTEGER, parseLong(value));
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: this should be `parseInt` I think

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java [hunk 13]
```diff
@@ -181,6 +188,9 @@ public static Object getField(Type type, Block block, int position)
         if (BigintType.BIGINT.equals(type)) {
             return type.getLong(block, position);
         }
+        if (IntegerType.INTEGER.equals(type)) {
+            return (int) (type.getLong(block, position));
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java]: the parentheses around `type.getLong` are unnecessary

File: presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java (hunk 4 above)
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java]: IntelliJ should warn you about this.  If it is not, ask @electrum to help you get it configured correctly.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java [hunk 14]
```diff
@@ -480,6 +492,16 @@ public static long bigintPartitionKey(String value, String name)
         }
     }
 
+    public static long intPartitionKey(String value, String name)
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: seems like this should return `int` rather than `long`

Comment (raghavsethi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java]: I'm modeling INT after DATE - i.e. it is also long backed.

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java (hunk 8 above)
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java]: That ends up calling into `readVInt` which already has a loop and a branch, so I would guess it's not a problem. I'd ask @dain's opinion though, since he has more experience with the readers

**Comment by cberner** [comment]
I see. If it's not standard then check with @martint that that's how we want it to work

**Comment by cberner** [comment]
Also, did you test this in the verifier?

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java [hunk 15]
```diff
@@ -324,7 +324,7 @@ public long getLong(int fieldId)
 
         if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(INTEGER) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
             // we don't use Preconditions.checkArgument because it requires boxing fieldId, which affects inner loop performance
-            throw new IllegalArgumentException(String.format("Expected field to be %s, %s or %s , actual %s (field %s)", BIGINT, DATE, TIMESTAMP, types[fieldId], fieldId));
+            throw new IllegalArgumentException(String.format("Expected field to be %s, %s, %s or %s , actual %s (field %s)", INTEGER, BIGINT, DATE, TIMESTAMP, types[fieldId], fieldId));
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java]: I think this is part of the wrong commit

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java [hunk 16]
```diff
@@ -217,13 +218,6 @@ private static boolean columnTypeSupported(List<HiveColumnHandle> columns)
...
                 .map(TypeSignature::getBase)
-                .allMatch(base -> INTEGER.equals(base) ||
-                        BIGINT.equals(base) ||
-                        BOOLEAN.equals(base) ||
-                        DOUBLE.equals(base) ||
-                        TIMESTAMP.equals(base) ||
-                        VARCHAR.equals(base) ||
-                        VARBINARY.equals(base) ||
-                        DATE.equals(base));
+                .allMatch(SUPPORTED_TYPES::contains);
```
Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java]: And this. In fact, maybe merge this commit with the previous one?

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java (hunk 15 above)
Comment (raghavsethi) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java]: Yep. This is my first stacked PR and I forgot to re-order my later fixes. I can merge before pushing.

**Comment by raghavsethi** [comment]
How would I test it in the verifier? Everything with INT would miscompare.. I've deployed a version to a test cluster and manually checked that reading, writing and operations work correctly.

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveFileFormats.java [hunk 17]
```diff
@@ -456,6 +465,9 @@ else if (testColumn.getObjectInspector().getTypeName().equals("date")) {
                         SqlDate expectedDate = new SqlDate(((Long) expectedValue).intValue());
                         assertEquals(actualValue, expectedDate, "Wrong value for column " + testColumn.getName());
                     }
+                    else if (testColumn.getObjectInspector().getTypeName().equals("int")) {
```
Comment (cberner) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveFileFormats.java]: what about small and tinyint?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveBucketing.java [hunk 18]
```diff
@@ -289,6 +289,7 @@ public static void appendToBlockBuilder(Type type, Object element, BlockBuilder
             case StandardTypes.BOOLEAN:
                 type.writeBoolean(blockBuilder, (Boolean) element);
                 break;
+            case StandardTypes.INTEGER:
```
Comment (cberner) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveBucketing.java]: passing `longValue()` doesn't seem correct for `INTEGER`

**Comment by cberner** [comment]
I think the verifier may work actually. I don't think it checks that the SQL type is the same, and compares all integral types as longs. Even if the correctness checks fail though, it would be good to make sure none of the queries fail due to missing functions

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 19]
```diff
@@ -254,6 +260,30 @@ public void createPartitionedTableAs(HiveStorageFormat storageFormat)
         assertFalse(queryRunner.tableExists(getSession(), "test_create_partitioned_table_as"));
     }
 
+    public void createPartitionedTableAsWithIntegerColumns(HiveStorageFormat storageFormat)
+            throws Exception
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_create_partitioned_table_as " +
```
Comment (cberner) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: this is the same table name used in the previous test, you need to change it to something else or this will fail when they run in parallel

File: presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java [hunk 20]
```diff
@@ -108,7 +108,12 @@ public static long modulus(@SqlType(StandardTypes.INTEGER) long left, @SqlType(S
     @SqlType(StandardTypes.INTEGER)
     public static long negate(@SqlType(StandardTypes.INTEGER) long value)
     {
-        return Math.negateExact((int) value);
+        try {
+            return Math.negateExact((int) value);
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java]: part of the wrong commit

File: presto-main/src/test/java/com/facebook/presto/type/TestIntegerOperators.java [hunk 21]
```diff
@@ -128,7 +128,7 @@ public void testNotEqual()
     }
 
     @Test
-    public void testessThan()
```
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/type/TestIntegerOperators.java]: usually we put cleanups in separate commits

File: presto-tests/src/main/java/com/facebook/presto/tests/TestingPrestoClient.java [hunk 22]
```diff
@@ -161,7 +162,7 @@ private static Object convertToRowValue(Type type, Object value, TimeZoneKey tim
         if (BOOLEAN.equals(type)) {
             return value;
         }
-        else if (BIGINT.equals(type)) {
+        else if (BIGINT.equals(type) || INTEGER.equals(type)) {
```
Comment (cberner) [inline presto-tests/src/main/java/com/facebook/presto/tests/TestingPrestoClient.java]: should this be `intValue` for `INTEGER`?

**Comment by cberner** [comment]
Back to you for comments

File: presto-parser/src/main/java/com/facebook/presto/sql/tree/IntegerLiteral.java [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+package com.facebook.presto.sql.tree;
+
+import com.facebook.presto.sql.parser.ParsingException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class IntegerLiteral
```
Comment (martint) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/IntegerLiteral.java]: As we discussed a few weeks ago, we shouldn't introduce a new AST node. The current `LongLiteral` should serve for all integer types (maybe with a change of name). Values should take the narrowest type the fit in (i.e., 1 => SMALLINT, 100000 => INTEGER, 100000000000 => BIGINT).

File: presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java (hunk 2 above)
Comment (raghavsethi) [inline presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java]: In ParquetPageSource there's not as much of a performance concern.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java (hunk 11 above)
Comment (raghavsethi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java]: Yup. Will do as soon as INT gets in.

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveFileFormats.java (hunk 17 above)
Comment (raghavsethi) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveFileFormats.java]: They were needed in checkCursor because they would incorrectly fall into the check for PRIMITIVE. They are not required here.

File: presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java (hunk 7 above)
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java]: Yes, this is fine.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveBucketing.java (hunk 18 above)
Comment (dain) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveBucketing.java]: type doesn't have `writeInteger`, so you have to pass in a long.  This is how the `Date` type works below.

**Comment by raghavsethi** [comment]
@cberner Addressed your comments.

File: presto-main/src/test/java/com/facebook/presto/type/TestIntType.java [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+
+public class TestIntType
```
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/type/TestIntType.java]: Since you used "Integer" in the test classes below, I would name this "TestIntegerType"

**Comment by cberner** [comment]
It looks like there are a lot of test failures in the product tests

File: presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java [hunk 25]
```diff
@@ -277,6 +277,9 @@ public WindowFunctionSupplier load(SpecializedFunctionKey key)
                     }
                 });
 
+        // If there are non-parametric functions here that have both Long and Double variants, they must be ordered such that the Long version comes first. If you do not do this,
+        // the integral types narrower than Long will be incorrectly coerced to Doubles. This will be fixed when we move to a lattice-based type coercion system.
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java]: Lets enforce this in `FunctionListBuilder`

**Comment by cberner** [comment]
One comment about the function ordering. Otherwise looks good

**Comment by raghavsethi** [comment]
@cberner Added another commit to fix the coercion issue. I will squash it with the reordering commit before merge.

**Comment by raghavsethi** [comment]
cc @martint for this commit: https://github.com/prestodb/presto/pull/4711/commits/3b69eb39387891ea3a0f7bc487fb3f65bb8cb71c

File: presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java [hunk 26]
```diff
@@ -467,12 +468,27 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
...
-                break;
+                coercedMatches.add(signature);
+            }
+        }
+
+        // TODO: remove when we move to a lattice-based type coercion system
+        if (coercedMatches.size() == 2) {
+            for (int i = 0; i < resolvedTypes.size(); i++) {
+                if (AMBIGUOUS_INTEGRAL_COERCION_SOURCES.contains(resolvedTypes.get(i)) &&
+                        typeManager.getType(coercedMatches.get(0).getArgumentTypes().get(i)).equals(DOUBLE) &&
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java]: Why do you need to call `getType` here? I think you can just replace it with `coercredMatches.get(0).getArgumentTypes().get(i).getBase().equals(StandardTypes.DOUBLE)`

File: presto-main/src/main/java/com/facebook/presto/metadata/Signature.java [hunk 27]
```diff
@@ -404,11 +404,16 @@ else if (canCoerce(boundParameters.get(parameter.getBase()), type) && typeParame
         if (!parametersMatched && !typeSignature.getBase().equals(parameter.getBase())) {
             // check for possible coercion
             if (allowCoercion && canCoerce(typeSignature, parameter)) {
+                Type coercedType = typeManager.getType(parseTypeSignature(parameter.getBase()));
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/metadata/Signature.java]: Not sure why you needed this change, but I think you can remove it, if you make the change in FunctionRegistry that I suggested

File: presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java [hunk 28]
```diff
@@ -156,11 +156,18 @@ private Type instantiateParametricType(TypeSignature signature)
         if (parametricType == null) {
             return null;
         }
-        Type instantiatedType = parametricType.createType(parameters);
 
-        // TODO: reimplement this check? Currently "varchar(Integer.MAX_VALUE)" fails with "varchar"
-        //checkState(instantiatedType.equalsSignature(signature), "Instantiated parametric type name (%s) does not match expected name (%s)", instantiatedType, signature);
-        return instantiatedType;
+        try {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java]: same comment

**Comment by raghavsethi** [comment]
@cberner also added the tempto update commit. Assuming product-tests passes now, am I good to go?

**Comment by cberner** [comment]
yep, so long as all the tests pass it looks good

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 29]
```diff
@@ -282,31 +281,6 @@ public void createPartitionedTableAs(HiveStorageFormat storageFormat)
         assertFalse(queryRunner.tableExists(getSession(), "test_create_partitioned_table_as"));
     }
 
-    public void createPartitionedTableAsWithIntegerColumns(HiveStorageFormat storageFormat)
```
Comment (cberner) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: Remove this from the commit where you added it? I assume this isn't in master

**Comment by cberner** [comment]
Looks good, but wait for the release before you land this
